TestVersionUpgradeToLatest7x is flaky for 7.1.1

[Recent failure](https://devops-ci.elastic.co/view/cloud-on-k8s/job/cloud-on-k8s-e2e-tests-stack-versions/130/testReport/junit/github/com_elastic_cloud-on-k8s_test_e2e_kb/Run_tests_for_different_ELK_stack_versions_in_GKE___7_1_1___TestVersionUpgradeToLatest7x_ES_cluster_health_should_eventually_be_green_01/) From version 7.1.1 to 7.8.0

```
=== RUN   TestVersionUpgradeToLatest7x/ES_cluster_health_should_eventually_be_green#01
Retries (5m0s timeout): ....................................................................................................
    TestVersionUpgradeToLatest7x/ES_cluster_health_should_eventually_be_green#01: utils.go:84: 
        	Error Trace:	utils.go:84
        	Error:      	Received unexpected error:
        	            	health is yellow
        	Test:       	TestVersionUpgradeToLatest7x/ES_cluster_health_should_eventually_be_green#01
{"log.level":"error","@timestamp":"2020-07-01T04:00:48.540Z","message":"stopping early","service.version":"0.0.0-00000000","service.type":"eck","ecs.version":"1.4.0","error":"test failure","error.stack_trace":"github.com/go-logr/zapr.(*zapLogger).Error\n\t/go/pkg/mod/github.com/go-logr/zapr@v0.1.0/zapr.go:128\ngithub.com/elastic/cloud-on-k8s/test/e2e/test.StepList.RunSequential\n\t/go/src/github.com/elastic/cloud-on-k8s/test/e2e/test/step.go:43\ngithub.com/elastic/cloud-on-k8s/test/e2e/test.RunMutationsWhileWatching\n\t/go/src/github.com/elastic/cloud-on-k8s/test/e2e/test/run_mutation.go:73\ngithub.com/elastic/cloud-on-k8s/test/e2e/kb.TestVersionUpgradeToLatest7x\n\t/go/src/github.com/elastic/cloud-on-k8s/test/e2e/kb/version_upgrade_test.go:51\ntesting.tRunner\n\t/usr/local/go/src/testing/testing.go:991"}
    --- FAIL: TestVersionUpgradeToLatest7x/ES_cluster_health_should_eventually_be_green#01 (300.00s)

```

Looking at `MasterService` logs: 


```
routing_nodes:
-----node_id[yaKIbhrSS8eADZEa_Q7yEA][V]
--------[data-integrity-check][1], node[yaKIbhrSS8eADZEa_Q7yEA], [P], s[STARTED], a[id=DIM8BbArRgm7Rlw7E1E8Ng]
--------[data-integrity-check][2], node[yaKIbhrSS8eADZEa_Q7yEA], [P], s[STARTED], a[id=gpsbrl0rRviXGT7vVwDi6w]
--------[data-integrity-check][0], node[yaKIbhrSS8eADZEa_Q7yEA], [P], s[STARTED], a[id=4y0athEsQmyCpd85MceRug]
--------[.kibana_task_manager][0], node[yaKIbhrSS8eADZEa_Q7yEA], [P], s[STARTED], a[id=D1int7cnRn2MIpoJDJUpoA]
--------[.kibana][0], node[yaKIbhrSS8eADZEa_Q7yEA], [P], s[STARTED], a[id=XXE9PruwTCeQ8o-vLyJDow]
--------[.kibana_1][0], node[yaKIbhrSS8eADZEa_Q7yEA], [P], s[STARTED], a[id=opER5-9DQuOjmnM8gJR8BA]
--------[.security-7][0], node[yaKIbhrSS8eADZEa_Q7yEA], [P], s[STARTED], a[id=aEHaPc7eSdqqkY_JXQhCkQ]
--------[.kibana_2][0], node[yaKIbhrSS8eADZEa_Q7yEA], [P], s[STARTED], a[id=h7Hjtc3cQeK7Dn7yYz-PWg]
---- unassigned
--------[.kibana][0], node[null], [R], recovery_source[peer recovery], s[UNASSIGNED], unassigned_info[[reason=INDEX_CREATED], at[2020-07-01T03:55:15.209Z], delayed=false, allocation_status[no_attempt]]

```

Cluster is never green because a `.kibana` index exists that should
* not be there in the first place but be just an `alias`
* should not have a replica on a single node cluster

It looks like the `.kibana` index is created immediately when we roll out Kibana (before even trying to upgrade) but after the creation of `.kibana_1`
```
cluster state updated, source [create-index [.kibana], cause [auto(bulk api)]]
cluster uuid: _dP5EuidTCKAtiff7bjKkQ [committed: true]
version: 41
state uuid: 1Wi25gTEQGacwx9m1cPzuQ
from_diff: false
meta data version: 39
   coordination_metadata:
      term: 1
      last_committed_config: VotingConfiguration{yaKIbhrSS8eADZEa_Q7yEA}
      last_accepted_config: VotingConfiguration{yaKIbhrSS8eADZEa_Q7yEA}
      voting tombstones: []
   [.kibana_task_manager/nnMiuNzATAGgxpwo7TO53Q]: v[5], mv[1], sv[2], av[1]
      0: p_term [1], isa_ids [D1int7cnRn2MIpoJDJUpoA]
   [.kibana/26Y91HpXSyyp_1aOJFDFIQ]: v[2], mv[1], sv[1], av[1]
      0: p_term [1], isa_ids []
   [.kibana_1/qerAJG6LSp-sLoMSBOvPzA]: v[4], mv[1], sv[2], av[1]
      0: p_term [1], isa_ids []
   [.security-7/_F4Owb_cRUG68wrGuwR6gA]: v[3], mv[1], sv[1], av[1]
      0: p_term [1], isa_ids [aEHaPc7eSdqqkY_JXQhCkQ]
metadata customs:
   ingest: org.elasticsearch.ingest.IngestMetadata@93e0dd94   index_lifecycle: {
  "policies" : {} 
   index-graveyard: IndexGraveyard[[]]
nodes: 
   {test-version-upgrade-to-7x-fq86-es-masterdata-0}{yaKIbhrSS8eADZEa_Q7yEA}{Bmq9xviyRvOWPLY1PRuFbg}{10.125.32.80}{10.125.32.80:9300}{dilmrt}{ml.machine_memory=2147483648, xpack.installed=true, transform.node=true, ml.max_open_jobs=20}, local, master
routing_table (version 7):
-- index [[.kibana_task_manager/nnMiuNzATAGgxpwo7TO53Q]]
----shard_id [.kibana_task_manager][0]
--------[.kibana_task_manager][0], node[yaKIbhrSS8eADZEa_Q7yEA], [P], s[STARTED], a[id=D1int7cnRn2MIpoJDJUpoA]

-- index [[.kibana/26Y91HpXSyyp_1aOJFDFIQ]]
----shard_id [.kibana][0]
--------[.kibana][0], node[yaKIbhrSS8eADZEa_Q7yEA], [P], recovery_source[new shard recovery], s[INITIALIZING], a[id=XXE9PruwTCeQ8o-vLyJDow], unassigned_info[[reason=INDEX_CREATED], at[2020-07-01T03:55:15.209Z], delayed=false, allocation_status[no_attempt]]
--------[.kibana][0], node[null], [R], recovery_source[peer recovery], s[UNASSIGNED], unassigned_info[[reason=INDEX_CREATED], at[2020-07-01T03:55:15.209Z], delayed=false, allocation_status[no_attempt]]

-- index [[.kibana_1/qerAJG6LSp-sLoMSBOvPzA]]
----shard_id [.kibana_1][0]
--------[.kibana_1][0], node[yaKIbhrSS8eADZEa_Q7yEA], [P], recovery_source[new shard recovery], s[INITIALIZING], a[id=opER5-9DQuOjmnM8gJR8BA], unassigned_info[[reason=INDEX_CREATED], at[2020-07-01T03:55:14.806Z], delayed=false, allocation_status[no_attempt]]

-- index [[.security-7/_F4Owb_cRUG68wrGuwR6gA]]
----shard_id [.security-7][0]
--------[.security-7][0], node[yaKIbhrSS8eADZEa_Q7yEA], [P], s[STARTED], a[id=aEHaPc7eSdqqkY_JXQhCkQ]

routing_nodes:
-----node_id[yaKIbhrSS8eADZEa_Q7yEA][V]
--------[.kibana_task_manager][0], node[yaKIbhrSS8eADZEa_Q7yEA], [P], s[STARTED], a[id=D1int7cnRn2MIpoJDJUpoA]
--------[.kibana][0], node[yaKIbhrSS8eADZEa_Q7yEA], [P], recovery_source[new shard recovery], s[INITIALIZING], a[id=XXE9PruwTCeQ8o-vLyJDow], unassigned_info[[reason=INDEX_CREATED], at[2020-07-01T03:55:15.209Z], delayed=false, allocation_status[no_attempt]]
--------[.kibana_1][0], node[yaKIbhrSS8eADZEa_Q7yEA], [P], recovery_source[new shard recovery], s[INITIALIZING], a[id=opER5-9DQuOjmnM8gJR8BA], unassigned_info[[reason=INDEX_CREATED], at[2020-07-01T03:55:14.806Z], delayed=false, allocation_status[no_attempt]]
--------[.security-7][0], node[yaKIbhrSS8eADZEa_Q7yEA], [P], s[STARTED], a[id=aEHaPc7eSdqqkY_JXQhCkQ]
---- unassigned
--------[.kibana][0], node[null], [R], recovery_source[peer recovery], s[UNASSIGNED], unassigned_info[[reason=INDEX_CREATED], at[2020-07-01T03:55:15.209Z], delayed=false, allocation_status[no_attempt]]
customs:
   security_tokens: TokenMetadata{ everything is secret }
```

--- a/test/e2e/kb/version_upgrade_test.go
+++ b/test/e2e/kb/version_upgrade_test.go
@@ -31,9 +31,16 @@ func TestVersionUpgradeToLatest7x(t *testing.T) {
 		WithESMasterDataNodes(1, elasticsearch.DefaultResources).
 		WithVersion(dstVersion)
 
+	srcNodeCount := 3
+	if srcVersion == "7.1.1" {
+		// workaround for https://github.com/elastic/kibana/pull/37674 to avoid accidental .kibana index creation
+		// can be removed once we stop supporting 7.1.1
+		srcNodeCount = 1
+	}
+
 	kbBuilder := kibana.NewBuilder(name).
 		WithElasticsearchRef(esBuilder.Ref()).
-		WithNodeCount(3).
+		WithNodeCount(srcNodeCount).
 		WithVersion(srcVersion)
 
 	opts := []client.ListOption{
@@ -51,7 +58,7 @@ func TestVersionUpgradeToLatest7x(t *testing.T) {
 	test.RunMutationsWhileWatching(
 		t,
 		[]test.Builder{esBuilder, kbBuilder},
-		[]test.Builder{esBuilder, kbBuilder.WithVersion(dstVersion).WithMutatedFrom(&kbBuilder)},
+		[]test.Builder{esBuilder, kbBuilder.WithVersion(dstVersion).WithNodeCount(3).WithMutatedFrom(&kbBuilder)},
 		[]test.Watcher{NewReadinessWatcher(opts...), NewVersionWatcher(opts...)},
 	)
 }

